packages/core-sdk/src/resources/royalty.ts

```
@@ -287,19 +363,35 @@ export class RoyaltyClient {
       });
       await this.rpcClient.waitForTransactionReceipt({ hash });
       txHashes.push(hash);
-
-      // auto unwrap WIP tokens once they are transferred
-      if (token === WIP_TOKEN_ADDRESS && !skipUnwrapIp) {
-        const withdrawalHash = await this.wrappedIpClient.withdraw({
-          value: amount,
-        });
-        txHashes.push(withdrawalHash);
-        await this.rpcClient.waitForTransactionReceipt({ hash: withdrawalHash });
-      }
     };
     for (const { token, amount } of claimedTokens) {
       await transferToken(token, amount);
     }
     return txHashes;
   }
+
+  private async getClaimerInfo(claimer: Address) {
+    const isClaimerIp = await this.ipAssetRegistryClient.isRegistered({
+      id: claimer,
+    });
+    const ipAccount = new IpAccountImplClient(this.rpcClient, this.wallet, claimer);
+    let ownsClaimer = claimer === this.walletAddress;
+    if (isClaimerIp) {
+      const ipOwner = await ipAccount.owner();
+      ownsClaimer = ipOwner === this.walletAddress;
+    }
+    return { ownsClaimer, isClaimerIp, ipAccount };
+  }
+
+  private async unwrapWipTokens(claimedTokens: IpRoyaltyVaultImplRevenueTokenClaimedEvent[]) {
+    const wipToken = claimedTokens.find((token) => token.token === WIP_TOKEN_ADDRESS);
```

lets be more defensive here. We should throw an error if there are multiple wip tokens or we can aggregate the values. I think aggregate might be a bit better?

---

While there is only one WIP token, the system might interact with other tokens, such as `ERC20`. Currently, the `unwrapWipTokens` method only focuses on handling `WIP`.  I don't want to interrupt user interaction when the token is another token instead of WIP.

---

I've made it clear you can aggregate the wip sum and unwrap. please make the change.

---

I think there might be some missing context here. The `claimedTokens` data is already aggregated, and its structure looks like this:

```json
[
  { token: WIP_TOKEN_ADDRESS, amount: 2n, claimer: walletAddress },
  { token: ERC20, amount: 0n, claimer: walletAddress }
]
```
This means `claimedTokens` does not represent `multiple WIP tokens`. Additionally, if the token is ERC20, should we throw an error? Let me know your thoughts!

---

since `unwrapWipTokens` is a util, its not guaranteed that that's passed in is aggregated. It could be right now but someone can use it elsewhere and pass an array with two WIP records.
You can do the following:
- Make sure there's no two WIP token records in `claimedTokens`, otherwise throw error. I've mentioned nothing about throwing an error if there's another erc20 in there.
- Aggregate WIP inside this function, to ensure we can never get two wip records.